Update docs to refine explanations for non-determinism (#4448)
temporalio/documentation: docs/develop/dotnet/workflows/basics.mdx

Temporal requires that the deterministic `TaskScheduler.Current` is used, but many .NET async calls will use `TaskScheduler.Default` implicitly (and some analyzers even encourage this).
Here are some known gotchas to avoid with .NET tasks inside of Workflows:

-  Use `Workflow.RunTaskAsync` instead of `Task.Run`. `Task.Run` uses the default scheduler and puts work on the thread pool.

>     - Can also use `Task.Factory.StartNew` with current scheduler or instantiate the `Task` and run `Task.Start` on it.

Reviewer: ```suggestion
    - You can also use `Task.Factory.StartNew` with current scheduler or instantiate the `Task` and run `Task.Start` on it.
```

- If you need to use `Task.ConfigureAwait`, use `Task.ConfigureAwait(true)`. `Task.ConfigureAwait(false)` won't use the current context.

>   - There is no significant performance benefit to `Task.ConfigureAwait` in workflows anyways due to how the scheduler works.

Reviewer: ```suggestion
  - There is no significant performance benefit to `Task.ConfigureAwait` in workflows because of how the scheduler works.
```
